backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/XmlDocumentExtractor.java

```diff
@@ -0,0 +1,167 @@
+package de.bund.digitalservice.ris.norms.application.service;
+
+import de.bund.digitalservice.ris.norms.application.service.exceptions.ModificationException;
+import de.bund.digitalservice.ris.norms.application.service.exceptions.XmlProcessingException;
+import java.util.Optional;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+/**
+ * Service class within the application core part of the backend. It is responsible for working with
+ * w3c documents. It is annotated with {@link Service} to indicate that it's a service component in
+ * the Spring context.
+ */
+@Service
+public class XmlDocumentExtractor {
```

Comment:
`...Extractor` looks very specific to me. It does much more than extracting.

As for a better name, I struggle to find one that's better than `...Helper` (as we can't just have a `...Functions` namespace).